[2025-08-14T21:54:51.779+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T21:46:15+00:00 [queued]>
[2025-08-14T21:54:51.791+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T21:46:15+00:00 [queued]>
[2025-08-14T21:54:51.792+0000] {taskinstance.py:2170} INFO - Starting attempt 3 of 3
[2025-08-14T21:54:51.816+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): fetch_historical_data> on 2025-08-14 21:46:15+00:00
[2025-08-14T21:54:51.821+0000] {standard_task_runner.py:60} INFO - Started process 32199 to run task
[2025-08-14T21:54:51.831+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'stock_market_batch_pipeline', 'fetch_historical_data', 'manual__2025-08-14T21:46:15+00:00', '--job-id', '49', '--raw', '--subdir', 'DAGS_FOLDER/stock_market_batch_dag.py', '--cfg-path', '/tmp/tmpd5pxa_d5']
[2025-08-14T21:54:51.837+0000] {standard_task_runner.py:88} INFO - Job 49: Subtask fetch_historical_data
[2025-08-14T21:54:51.980+0000] {task_command.py:423} INFO - Running <TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T21:46:15+00:00 [running]> on host 0181ae2045ff
[2025-08-14T21:54:52.088+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Basanth Kumar' AIRFLOW_CTX_DAG_ID='stock_market_batch_pipeline' AIRFLOW_CTX_TASK_ID='fetch_historical_data' AIRFLOW_CTX_EXECUTION_DATE='2025-08-14T21:46:15+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-14T21:46:15+00:00'
[2025-08-14T21:54:52.090+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-08-14T21:54:52.093+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n    unset PYTHONNOUSERSITE\n    export PYTHONPATH=/home/***/.local/lib/python3.8/site-packages\n    cd /opt/***/dags/scripts\n    python batch_data_producer.py 2025-08-14\n    ']
[2025-08-14T21:54:52.111+0000] {subprocess.py:86} INFO - Output:
[2025-08-14T21:54:52.958+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-08-14T21:54:52.970+0000] {subprocess.py:93} INFO -   File "batch_data_producer.py", line 8, in <module>
[2025-08-14T21:54:52.971+0000] {subprocess.py:93} INFO -     import yfinance as yf
[2025-08-14T21:54:52.975+0000] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'yfinance'
[2025-08-14T21:54:53.145+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-08-14T21:54:53.191+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-08-14T21:54:53.196+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=stock_market_batch_pipeline, task_id=fetch_historical_data, execution_date=20250814T214615, start_date=20250814T215451, end_date=20250814T215453
[2025-08-14T21:54:53.236+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 49 for task fetch_historical_data (Bash command failed. The command returned a non-zero exit code 1.; 32199)
[2025-08-14T21:54:53.268+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-08-14T21:54:53.362+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
